python-avd/pyavd/_eos_designs/structured_config/underlay/vlans.py

```diff
@@ -50,3 +47,17 @@ def vlans(self: AvdStructuredConfigUnderlayProtocol) -> None:
         )
         for peer_uplink_native_vlan in uplink_native_vlans:
             self.structured_config.vlans.append_new(id=int(peer_uplink_native_vlan), name="NATIVE", state="suspend")
+
+    def _update_underlay_vlan_trunk_groups(self: AvdStructuredConfigUnderlayProtocol, vlans: EosCliConfigGen.Vlans) -> None:
+        """Update trunk groups to configure on the underlay link."""
+        if self.inputs.enable_trunk_groups is not True:
+            return
+
+        for peer in self._avd_peers:
+            peer_facts = self.shared_utils.get_peer_facts(peer, required=True)
+            for uplink in peer_facts["uplinks"]:
+                if uplink["peer"] == self.shared_utils.hostname:
+                    if (peer_trunk_groups := get(uplink, "peer_trunk_groups")) is None:
+                        continue
+                    for vlan in range_expand(uplink["vlans"]):
+                        vlans.obtain(int(vlan)).trunk_groups.extend(peer_trunk_groups)
```

Comment:
maybe we need append_unique here to make sure we don't put multiple time the same value.

@ClausHolbechArista

Reply:
Yes.

Reply:
The schema could probably be improved to have unique and then our classes could check for that and automatically do append_unique if set.

Reply:
using append_unique for now